docs/content/concepts/partitions-schedules-sensors/partitions.mdx

```diff
@@ -133,6 +134,49 @@ result = materialize(
 )
 ```
 
+#### Dynamically partitioned assets (Experimental)
+
+<Note>
+  This feature is <strong>experimental</strong>. Until this feature is marked as
+  stable, partition created prior may not be retained.
+</Note>
+
+The <PyObject object="DynamicPartitionsDefinition" /> class accepts a name argument, representing the name of the partition set:
+
+```python file=/concepts/partitions_schedules_sensors/dynamic_partitioned_asset.py startafter=start_dynamic_partitions_marker endbefore=end_dynamic_partitions_marker
+images_partitions_def = DynamicPartitionsDefinition(name="images")
+
+
+@asset(partitions_def=images_partitions_def)
+def images(context):
+    ...
+```
+
+For a given dynamic partition set, partition keys can be added and removed. One common pattern is detecting the presence of a new partition through a [sensor](concepts/partitions-schedules-sensors/sensors#sensors), adding the partition and then triggering a run for that partition:
+
+```python file=/concepts/partitions_schedules_sensors/dynamic_partitioned_asset.py startafter=start_dynamic_partitions_2 endbefore=end_dynamic_partitions_2
+images_job = define_asset_job(
+    "images_job", AssetSelection.keys("images"), partitions_def=images_partitions_def
+)
+
+
+@sensor(job=images_job)
+def image_sensor(context):
+    new_images = [
+        img_filename
+        for img_filename in os.listdir(os.getenv("MY_DIRECTORY"))
+        if not images_partitions_def.has_partition(img_filename, context.instance)
```

Comment:
This makes me think that we should definitely introduce a `has_partitions` method.

Reply:
👍 👍